File: 10_기타알고리즘/08_소수구하기.cpp
#include <iostream>
#include <cmath>
#include <cstring>

using namespace std;

void printPrimeNums(bool* primes, int m, int n){
    for (int i = 2; i <= sqrt(n); i++)
    {
        if(primes[i]){
            int j = 2;
            while(1){
                if(i * j > n){
                    break;
                }
                primes[i * j] = false;
                j += 1;
            }
        }
    }
    for (int i = m; i <= n; i++)
    {
        if(primes[i]){
            cout << i << '\n';
            // endl은 출력 버퍼를 지워야 해서 더 느리다.
            // endl을 사용했을 때 시간 초과를 받았다.
        }
    }
}

int main(void) {

    int m, n;
    cin >> m >> n;
    
    bool* prime = new bool[n+1];
    memset(prime, true, sizeof(bool) * (n+1));
    prime[1] = false;

    printPrimeNums(prime, m, n);

    return 0;
}

/*
3 16
*/

File: 10_기타알고리즘/08_소수구하기.py

import math

m, n = map(int, input().split())
prime = [True for i in range(n+1)]
# 1은 소수가 아니기 때문에 무조건 초기화 False해줘야 한다. (반복문을 2부터 실행시키기 때문)
prime[1] = False

for i in range(2, int(math.sqrt(n)) + 1) :
    if prime[i] == True :
        j = 2
        while True :
            if i*j > n :
                break
            prime[i * j] = False
            j += 1

for i in range(m , n+1) :
    if prime[i] == True :
        print(i)

'''

3 16

'''